docs/source/advanced-usage.rst

```diff
@@ -0,0 +1,68 @@
+Advanced Usage
+==============
+
+This document explains some of the more advanced usage concepts with
+`wsproto`. This is assume you are familiar with `wsproto` and I/O in
+Python.
+
+Back-pressure
+-------------
+
+Back-pressure is an important concept to understand when implementing a
+client/server protocol. This section briefly explains the issue and then
+explains how to handle back-pressure when using `wsproto`.
+
+Imagine that you have a WebSocket server that reads messages from the client,
+does some processing, and then sends a response. What happens if the client
+sends messages faster than the the server can process them? If the incoming
+messages are buffered in memory, then the server will slowly use more and more
+memory, until the OS eventually kills it. This scenario is directly applicable
+to `wsproto`, because every time you call ``receive_data()``, it appends that
```

Comment:
Please use `receive_data(some_byte_string_of_data)`.
Writing it with just `()` is a bit misleading.

Reply:
Done.